microsoft/surface/repos.nix

```diff
@@ -1,8 +1,27 @@
-{ lib, pkgs, fetchgit }:
+{ lib, pkgs, fetchgit, fetchurl }:
 {
   linux-surface = fetchgit {
     url="https://github.com/linux-surface/linux-surface.git";
     rev="25ab2cf75e5eda5ab9739db1907300010c06dacf";
     sha256="0h8624d7ix1p6ysw9bllmnnwnv164z8xkx56zj3vdczn91vmqcf9";
   };
+
+  linux-kernel-firmware = fetchgit {
+    url="https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git";
+    rev="060ad8b3d505e2ff8878d11a81da3532b983cd45";
+    sha256="0n93xc3vzx7gy96lpkypw7ay201x7whirxm00c0zhq6zmq2zs12v";
+  };
+
+  ath10k-firmware = fetchgit {
+    url="https://github.com/kvalo/ath10k-firmware";
+    rev="84b47062aab31d67156e0a3ef593a6999a12864b";
+    sha256="0l8wfj8z4jbb31nzqkaxisby0n6061ix01c5di9bq66iby59j8py";
+  };
+
+    surface-go-ath10k-firmware = fetchurl {
+    url="https://support.killernetworking.com/K1535_Debian/board.bin";
+    # url="https://www.killernetworking.com/support/K1535_Debian/board.bin";
```

Comment:
Is see.

Reply:
This looks like it could break anytime the firmware is upgraded. Do they have versioned downloads?

Reply:
> This looks like it could break anytime the firmware is upgraded. Do they have versioned downloads?

Not as far as I can tell.
And their Linux docs only say "Use the official firmware repo for Linux" — which is the one that has the non-working firmware.

Every other post of wiki entry I could find for other OSes (e.g. Arch, Debian) has the same advice for d/l directly from Killer Networking.

There appear to be some efforts to get this firmware included in the official firmware, but it seems to have stalled.

This draft PR has the `board.bin` embedded in it, for testing.
Once I'm happy with how the `.nix` code is looking, I plan to figure out how to get a stable version of the firmware BLOB.... which I think I've neglected to add to `TODO.org` (will fix that)

Reply:
Should we upload the firmware to archive.org instead? Unversioned links is a bit of a minefield.

Reply:
I hadn't thought of that.
Is there a risk of this dropping off, too?

This is what I had thought:

- I suggested that the GitHub linux-surface owners might like to put it in a repo of their own approx. a week ago; I've been quite busy, so haven't had time to follow that up.

- I tried to follow the email thread to request the upstream `ath10k` owner add this `board.bin` but it's complete radio silence, AFAICT.
  I'm still hoping that sometime soon, they either respond to the email request to add it, or shares info about how to create the aggregate `board-2.bin` firmware blob.

- Should I create a public repo of my own, and add it there? This feels a bit like the last option, though.

Reply:
All options sounds acceptable to me.

Reply:
> All options sounds acceptable to me.

Thanks.

The upstream team are talking about my suggestion to put the firmware in their own repo, but also whether they should combine some of their other firmware repo's into a single one.
Until that's decided/done I'm going to use the copy in my own repo — primarily to unstick this PR.